databricks/koalas/indexes.py

```diff
@@ -961,6 +961,81 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def delete(self, loc):
+        """
+        Make new Index with passed location(-s) deleted.
+
+        .. note:: this API can be pretty expensive since it is based on
+             a global sequence internally.
+
+        Returns
+        -------
+        new_index : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
+        >>> kidx
+        Int64Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete(0)
+        Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete([0, 1, 2, 3, 10, 11])
+        Int64Index([4, 2, 4, 4, 2, 2], dtype='int64')
+        """
+        if isinstance(self, MultiIndex):
+            raise NotImplementedError("currently doesn't support for MultiIndex")
+        if not is_list_like(loc):
+            if not isinstance(self, str) and abs(loc) >= len(self):
```

Comment:
`len(self)` is non-trivial because it triggers a job to count all items. Can you call it once instead of calling twice?